llvm/test/CodeGen/AIE/aie2/schedule/resource/memory_bank.mir

```diff
@@ -0,0 +1,1119 @@
+    ; CHECK-NEXT: $wl2 = VLDA_dmw_lda_w_ag_idx_imm killed $p0, 0 :: (load (<8 x s32>), addrspace 14)
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 11)
+    $wl2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 14)
+...
+
+---
+name:            VLDA_VLDB_different_bank_BC_BD
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: VLDA_VLDB_different_bank_BC_BD
+    ; CHECK: $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 12)
+    ; CHECK-NEXT: $wl2 = VLDA_dmw_lda_w_ag_idx_imm killed $p0, 0 :: (load (<8 x s32>), addrspace 13)
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 12)
+    $wl2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 13)
+...
+
+---
+name:            VLDA_VLDB_different_bank_BC_CD
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: VLDA_VLDB_different_bank_BC_CD
+    ; CHECK: $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 12)
+    ; CHECK-NEXT: $wl2 = VLDA_dmw_lda_w_ag_idx_imm killed $p0, 0 :: (load (<8 x s32>), addrspace 14)
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 12)
+    $wl2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 14)
+...
+
+
+
+---
+name:            VLDA_VLDB_different_bank_ABCD
+alignment:       16
+body:             |
+  bb.0.entry:
+    ; CHECK-LABEL: name: VLDA_VLDB_different_bank_ABCD
+    ; CHECK: BUNDLE implicit-def $wl2, implicit-def $wl1, implicit $p0 {
+    ; CHECK-NEXT:   $wl2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 6)
+    ; CHECK-NEXT:   $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 5)
+    ; CHECK-NEXT: }
+    ; CHECK-NEXT: BUNDLE implicit-def $wh2, implicit-def $wh1, implicit killed $p0 {
+    ; CHECK-NEXT:   $wh2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 8)
+    ; CHECK-NEXT:   $wh1 = VLDB_dmw_ldb_ag_idx_imm killed $p0, 0 :: (load (<8 x s32>), addrspace 7)
+    ; CHECK-NEXT: }
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
+    $wl1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 5)
+    $wl2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 6)
+    $wh1 = VLDB_dmw_ldb_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 7)
+    $wh2 = VLDA_dmw_lda_w_ag_idx_imm $p0, 0 :: (load (<8 x s32>), addrspace 8)
+...
```

Comment:
Can you add some test cases for TM loads? E.g.
 - One VLDB without MMO and one LD.TM with TileMemory MMO
 - One VLDB with addrspace MMO and one LD.TM with TileMemory MMO

For both cases I believe we should be able to schedule the two loads in the same cycle.

Reply:
These tests are added in  `llvm/test/CodeGen/AIE/aie2/schedule/memory_access_DM_TM.mir `